### PR TITLE
Added readme and auth conflicts docs

### PR DESCRIPTION
**Pull request by kirthik**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@kirthik__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by hovsepm** [comment]
@kirthik please pull from upstream

File: Documentation/AuthConflictsBetweenSDKs.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Bridging gap between old and new SDKs
+
+Authentication conflicts can happen due to mixing new and old azure SDKs. The credential types created by the new authentication library (Microsoft.Rest.ClientRuntime.Azure.Authentication, type = ServiceClientCredentials)  are incompatible with the credential types  used in older SDKs (Microsoft.Azure.Common, type = SubscriptionCloudCredentials). The problem is that using two different authentication libraries will require you to authenticate twice, which is a painful experience.
```
Comment (markcowl) [inline Documentation/AuthConflictsBetweenSDKs.md]: Need to flesh out what is meant by new and old SDKs.  You can list the old SDKs, and specify that SDKs with depend on Microsoft.Azure.Common are old and SDKs which depend on Microsoft.Rest.ClientRuntime are new

File: Documentation/AuthConflictsBetweenSDKs.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Bridging gap between old and new SDKs
+
+Authentication conflicts can happen due to mixing new and old azure SDKs. The credential types created by the new authentication library (Microsoft.Rest.ClientRuntime.Azure.Authentication, type = ServiceClientCredentials)  are incompatible with the credential types  used in older SDKs (Microsoft.Azure.Common, type = SubscriptionCloudCredentials). The problem is that using two different authentication libraries will require you to authenticate twice, which is a painful experience.   
+
+## Recommended solution
+To bridge this gap, the provided adapter allows you to use the new credentials with clients that require the older credential type.
```
Comment (markcowl) [inline Documentation/AuthConflictsBetweenSDKs.md]: the following adapter

File: Documentation/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+- Key Vault
+- Operational Insights
+- Recovery Services
+- Search
+- Server Management
+- Site Recovery
+- SQL
+- Stream Analytics
+
+Please look at [Azure documentation](https://azure.microsoft.com/en-us/documentation/api/) for more details
```
Comment (markcowl) [inline Documentation/README.md]: look at -> see

**Comment by markcowl** [comment]
@kirthik please pull the latest as waell - travis failure is at the tip of the AutoRest branch

File: Documentation/README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+- Redis Cache
+- Resource Manager (ARM)
+
+The following services use Hyak
+- Azure Backup
+- Azure Stack
+- Data Factory
+- Datalake AnalyticsCatalog
+- Datalake AnalyticsJob
+- DataLake StoreFileSystem
```
Comment (hovsepm) [inline Documentation/README.md]: 3 DataLake services are deprecated. They have been moved to Swagger/Autorest and renamed to DataLake Analytics and DataLake Store

File: Documentation/README.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# Microsoft Azure SDK for .NET Documentation
+
+The following services use AutoRest
+- AppServices
+- Compute
+- DataLake Analytics
+- DataLake Store
+- DataLake Storeuploader
```
Comment (hovsepm) [inline Documentation/README.md]: This is hand written code and does not have anything generated.

File: Documentation/README.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+- Datalake AnalyticsCatalog
+- Datalake AnalyticsJob
+- DataLake StoreFileSystem
+- HdInsight
+- Insights
+- Intune
+- Key Vault
+- Operational Insights
+- Recovery Services
+- Search
```
Comment (hovsepm) [inline Documentation/README.md]: Search is Autorest based

File: Documentation/README.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+- Server Management
```
Comment (hovsepm) [inline Documentation/README.md]: this is AutoRest generated but not CoreCLR ;)

File: Documentation/README.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+The following services use Hyak
+- Azure Backup
+- Azure Stack
+- Data Factory
+- Datalake AnalyticsCatalog
+- Datalake AnalyticsJob
+- DataLake StoreFileSystem
+- HdInsight
+- Insights
+- Intune
```
Comment (hovsepm) [inline Documentation/README.md]: AutoRest based but not CoreCLR

File: Documentation/README.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+- Compute
+- DataLake Analytics
+- DataLake Store
+- DataLake Storeuploader
+- DevTestLabs
+- Identity
+- Machine Learning
+- Networking
+- NotificationHubs
+- PowerBI
```
Comment (hovsepm) [inline Documentation/README.md]: Logic and Scheduler are also AutoRest based

File: Documentation/README.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
+# Microsoft Azure SDK for .NET Documentation
+
+The following services use AutoRest
+- AppServices
```
Comment (hovsepm) [inline Documentation/README.md]: what is AppServices?

File: Documentation/README.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+- DataLake Store
+- DataLake Storeuploader
+- DevTestLabs
+- Identity
+- Machine Learning
+- Networking
+- NotificationHubs
+- PowerBI
+- Redis Cache
+- Resource Manager (ARM)
```
Comment (hovsepm) [inline Documentation/README.md]: I think you are missing some. Here is the full list of AutoRest generated and CoreCLR migrated libraries:
```
Microsoft.Azure.Graph.RBAC                             
Microsoft.Azure.Management.Authorization               
Microsoft.Azure.Management.Cdn                         
Microsoft.Azure.Management.CognitiveServices           
Microsoft.Azure.Management.Compute                     
Microsoft.Azure.Management.DataLake.Analytics          
Microsoft.Azure.Management.DataLake.Store              
Microsoft.Azure.Management.DevTestLabs                 
Microsoft.Azure.Management.Dns                         
Microsoft.Azure.Management.MachineLearning             
Microsoft.Azure.Management.Network                     
Microsoft.Azure.Management.NotificationHubs            
Microsoft.Azure.Management.PowerBIEmbedded             
Microsoft.Azure.Management.Redis                    
Microsoft.Azure.Management.ResourceManager            
Microsoft.Azure.Management.Storage                    
Microsoft.Azure.Management.TrafficManager             
Microsoft.Azure.Management.Websites                    
```

File: Documentation/README.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
+# Microsoft Azure SDK for .NET Documentation
+
+The following services use AutoRest
+- AppServices
+- Compute
+- DataLake Analytics
+- DataLake Store
+- DataLake Storeuploader
+- DevTestLabs
+- Identity
```
Comment (hovsepm) [inline Documentation/README.md]: Identity? I think we call it Authorization

File: Documentation/AuthConflicsBetweenSDKs.md [hunk 13]
```diff
@@ -0,0 +1,47 @@
+# Bridging gap between old and new SDKs
```
Comment (hovsepm) [inline Documentation/AuthConflicsBetweenSDKs.md]: you have 3 files in the PR:
```
1. Documentation/AuthConflicsBetweenSDKs.md
2. Documentation/AuthConflictsBetweenSDKs.md
3. Documentation/README.md
```

What is the difference between 1 and 2, except file name typo?

Comment (kirthik) [inline Documentation/AuthConflicsBetweenSDKs.md]: No difference. Din't notice a new file got added after I fixed the typo in file name. I will remove one

File: Documentation/README.md (hunk 10 above)
Comment (kirthik) [inline Documentation/README.md]: Websites. I will change to websites

**Comment by kirthik** [comment]
@hovsepm thanks for your review. I updated based on your feedback.

**Comment by hovsepm** [comment]
LGTM

**Pull request merged**